halyard-cli/src/main/java/com/netflix/spinnaker/halyard/cli/command/v1/config/security/authn/oauth2/EditOAuth2Command.java

```diff
@@ -19,15 +19,16 @@
 import com.beust.jcommander.DynamicParameter;
 import com.beust.jcommander.Parameter;
 import com.beust.jcommander.Parameters;
-import com.netflix.spinnaker.halyard.cli.command.v1.config.security.authn.AbstractEditAuthnMethodCommand;
-import com.netflix.spinnaker.halyard.cli.command.v1.converter.OAuth2ProviderTypeConverter;
-import com.netflix.spinnaker.halyard.config.model.v1.security.AuthnMethod;
+import com.netflix.spinnaker.halyard.cli.command.v1.config.AbstractConfigCommand;
+import com.netflix.spinnaker.halyard.cli.services.v1.Daemon;
+import com.netflix.spinnaker.halyard.cli.services.v1.OperationHandler;
 import com.netflix.spinnaker.halyard.config.model.v1.security.OAuth2;
-import lombok.Getter;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
 
 @Parameters(separators = "=")
-public class EditOAuth2Command extends AbstractEditAuthnMethodCommand<OAuth2> {
-  @Getter private AuthnMethod.Method method = AuthnMethod.Method.OAuth2;
+public class EditOAuth2Command extends AbstractConfigCommand {
```

Comment:
I could use an education here.  Why is AbstractConfigCommand correct instead of AbstractEditAuthnMethodCommand?

Reply:
`AbstractEditAuthnMethodCommand` sets the OAuth2 object under `Security.Authn.OAuth2`, but with the new structure, it should be under `Spring.Security.OAuth2`. We used `AbstractConfigCommand` because all commands in the `ConfigCommand` flow should extend it. Since `AbstractEditAuthnMethodCommand` already extends `AbstractConfigCommand`, the structure is consistent.